pulsar-function-go/pf/instance.go

```diff
@@ -0,0 +1,265 @@
+		if consumerConf.ReceiverQueueSize != nil {
+			if consumerConf.IsRegexPattern {
+				consumer, err = gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:     topic,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              subscriptionType,
+					MessageChannel:    channel,
+				})
+			} else {
+				consumer, err = gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:             topic,
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              subscriptionType,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					MessageChannel:    channel,
+				})
+			}
+		} else {
+			if consumerConf.IsRegexPattern {
+				consumer, err = gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:    topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             subscriptionType,
+					MessageChannel:   channel,
+				})
+			} else {
+				consumer, err = gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:            topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             subscriptionType,
+					MessageChannel:   channel,
+				})
+
+			}
+		}
+
+		if err != nil {
+			log.Errorf("create consumer error:%s", err.Error())
+			return nil, err
+		}
+		gi.consumers[topic] = consumer
+		gi.context.InputTopics = append(gi.context.InputTopics, topic)
+	}
+	return channel, nil
+}
+
+func (gi *GoInstance) handlerMsg(input pulsar.Message) (output []byte, err error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = NewContext(ctx, gi.context)
+	msgInput := input.Payload()
+	return gi.function.Process(ctx, msgInput)
+}
+
+func (gi *GoInstance) processResult(msgInput pulsar.Message, output []byte) {
+	atMostOnce := gi.context.InstanceConf.FuncDetails.ProcessingGuarantees == pb.ProcessingGuarantees_ATMOST_ONCE
+	autoAck := gi.context.InstanceConf.FuncDetails.AutoAck
+
+	if output != nil && gi.context.InstanceConf.FuncDetails.Sink.Topic != "" {
+		asyncMsg := pulsar.ProducerMessage{
+			Payload: output,
+		}
+		// Attempt to send the message asynchronously and handle the response
+		gi.producer.SendAsync(context.Background(), asyncMsg, func(message pulsar.ProducerMessage, e error) {
+			if e != nil {
+				log.Fatal(e)
+				gi.nackInputMessage(msgInput)
+			} else if !atMostOnce && autoAck {
+				gi.ackInputMessage(msgInput)
+			}
+		})
+	} else {
+		if autoAck {
+			gi.ackInputMessage(msgInput)
+		}
+	}
+}
+
+// ackInputMessage doesn't produce any result or the user doesn't want the result.
+func (gi *GoInstance) ackInputMessage(inputMessage pulsar.Message) {
+	gi.consumers[inputMessage.Topic()].Ack(inputMessage)
+}
+
+func (gi *GoInstance) nackInputMessage(inputMessage pulsar.Message) {
```

Comment:
why is this commented?

Reply:
Because in the current version of pulsar-client-go, we do not support this operation. This feature will be implemented in the next pull request.